cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocUtils.java

```diff
@@ -174,8 +175,16 @@ public static Map<String, String> parseKeyValueConfig(@Nullable String configVal
       return map;
     }
     for (String property : configValue.split(delimiter)) {
+      // Ignore empty property strings.
+      if (property.trim().isEmpty()) {
+        continue;
+      }
       String[] parts = property.split(keyValueDelimiter, 2);
       String key = parts[0].trim();
+      if (key.isEmpty()) {
+        throw new IllegalArgumentException(
+            "Empty key in property " + property);
+      }
       String value = parts.length > 1 ? parts[1].trim() : "";
```

Comment:
Restored the previous logic for parsing common labels from cdap-site.

Reply:
the comment there seems to primarily about using the separators used (`|` and `;`). 

The important thing behavior wise is that we shouldn't break backwards compatibility, and when somebody uses an invalid label, there should be an easy way to tell what was wrong. So it would be better to add arguments for the separators to the parseLabels method, and to use that method when getting both sets of labels.